### PR TITLE
feat: deploy to cloudflare pages

### PR DESCRIPTION
**Pull request by MantisClone**
Fixes [dw-234](https://app.dework.xyz/banklessdao-1/website-project-2?taskId=951d6660-0128-4bb0-bad7-68f1bb32c187)

# Review Instructions

Ignore the Vercel Preview. Instead, view the new Cloudflare Pages deployment.

# Production Deployment

https://github-pages.bankless-website.pages.dev/

# Changes

* Fixed Formatting (Most of the changes are this.  Sorry)
* Configure next.config.js
  * Enable static exports
  * Disable server-based image optimization
* Upgrade:
  * Next.js 13.5.4
  * Node 
  * Chakra-ui/react 2.8.1
  * Typescript eslint 6.7.4
  * React 18.2.0
  * eslint 8.50.0
  * Yarn 3.6.3
* Remove `@web3-onboard/keystone` dependency
* Ignore most of the .yarn/ directory except the yarn-3.6.3.cjs file
* Make pre-commit executable
* Delete unused files:
  * Next.js template.tsx
  * TodoForm.tsx
  * api/hello.ts
* Remove unused imports and variables
* Fixed Type errors

### CODE REVIEW

**Comment by MantisClone** [comment]
I successfully deployed, but the routing doesn't work and none of the images show up :sob: 

Deployed: https://banklessdao.github.io/bankless-website/
Github Action: https://github.com/BanklessDAO/bankless-website/actions/runs/6210975369

**Comment by MantisClone** [comment]
@BogDrakonov @didierkrux Any idea what I'm doing wrong?

**Comment by didierkrux** [comment]
> @BogDrakonov @didierkrux Any idea what I'm doing wrong?

I would try this instead: `src='images/logo.png'`

**Comment by didierkrux** [comment]
> > @BogDrakonov @didierkrux Any idea what I'm doing wrong?
> 
> I would try this instead: `src='images/logo.png'`

here is how I would have done it in the first place:
`basePath: "/bankless-website/",`
+ keep `src='/images/logo.png'`

**Comment by MantisClone** [comment]
Oops. Sorry, just realized there are a few links still broken.  I'll re-request the review after they're fixed.

File: next.config.js [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+   * Disable server-based image optimization. Next.js does not support
+   * dynamic features with static exports.
+   *
+   * @see https://nextjs.org/docs/pages/api-reference/components/image#unoptimized
+   */
   images: {
-    domains: ['s3-alpha-sig.figma.com']
-}
+    unoptimized: true,
+  }
```
Comment (MantisClone) [inline next.config.js:29]: Here's the important part that makes the static export and Github Pages work.

File: components/home/section2.tsx [hunk 2]
```diff
@@ -7,26 +7,25 @@ import Underline from 'components/_common/underline'
 
 const CARDS = [
   {
-    img: '/images/coming-home.jpg',
+    img: '/bankless-website/images/coming-home.jpg',
     title: 'Departments',
     text: 'Departments are operation-critical units of BanklessDAO without which the DAO cannot function properly.',
     href: '/about-us/departments',
```
Comment (MantisClone) [inline components/home/section2.tsx:13]: Interestingly, the `href`'s in the `CARDS` list DON'T need the `/bankless-website` prefix, else they end up wrong `/bankless-website/bankless-website/about-us/departments`. :shrug: No clue why, but it's fixed now.

File: assets/content/pages/work-with-us/carouselContent.json [hunk 3]
```diff
@@ -182,11 +182,11 @@
           "Let us know what you are looking for and we'll sync you with the right team."],
         "cta": {
           "text": "Visit BanklessDAO Podcasts",
-          "link": "https://www.bankless.community/listen",
+          "link": "/listen",
```
Comment (MantisClone) [inline assets/content/pages/work-with-us/carouselContent.json:185]: I fixed this internal link. It didn't work when I added the `/bankless-website` prefix. :shrug:  not sure why.

File: components/about-us/introduction.tsx [hunk 4]
```diff
@@ -23,8 +23,8 @@ const keyframeCircleMove = keyframes`
 `
 
 const keyframeBeat = keyframes`
-	to { 
-    transform: scale(2) 
+	to {
+    transform: scale(2)
```
Comment (MantisClone) [inline components/about-us/introduction.tsx:27]: Minor formatting fix.

File: package.json [hunk 5]
```diff
@@ -39,7 +39,7 @@
     "cypress-image-snapshot": "^4.0.1",
     "framer-motion": "^5.6.0",
     "lodash": "^4.17.21",
-    "next": "12.1.0",
+    "next": "^12.3.4",
```
Comment (MantisClone) [inline package.json:42]: Had to upgrade to Next 12.3.4 to get the `static-export` feature.

File: public/.nojekyll
Comment (MantisClone) [inline public/.nojekyll:1]: This file is required to prevent github-pages from running static content trough jekyll

**Comment by MantisClone** [comment]
I spent a few hours today trying to migrate to Cloudflare Pages.  Doing so seems to require updating many of our dependencies (Next, Node, Chakra-ui, Typescript, and even Yarn) to fit with the modern Cloudflare environment. Cloudflare only offers two environments: one that is very modern with Yarn v3, and one that seems weirdly old (with Node v12).

Finally, it deployed successfully. (The URLs are all messed up because they're edited to work on Github Pages, but it deployed so that's progress)
https://github-pages.bankless-website.pages.dev/

File: .husky/pre-commit
Comment (MantisClone) [inline .husky/pre-commit:1]: Make pre-commit executable.

File: .gitignore [hunk 6]
```diff
@@ -45,4 +45,14 @@ yarn-error.log*
...
+
+# yarn
+
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (MantisClone) [inline .gitignore:58]: Ignore *most* of the .yarn/ directory except the `yarn-3.6.3.cjs` file

**Review by didierkrux — APPROVED** [review]
Looks good to me 👍
Great job on the cloudflare implementation + fixing all the compatibilities issue 😅

**Comment by MantisClone** [comment]
Deploying! :crossed_fingers: 
image

**Comment by MantisClone** [comment]
Deployed! :tada: 
https://7c0ef1ed.bankless-website.pages.dev/

**Comment by BogDrakonov** [comment]
> Cloudflare Pages Deploying! 🤞 image

Do you need us to adjust DNS for this?

**Pull request merged**